kmk/modules/potentiometer.py

```diff
@@ -0,0 +1,88 @@
+from analogio import AnalogIn
+from supervisor import ticks_ms
+
+from kmk.modules import Module
+
+
+class Potentiometer:
+    def __init__(self, pin, move_callback, is_inverted=False):
+        self.is_inverted = is_inverted
+        self.read_pin = AnalogIn(pin)
+        self._direction = None
+        self._pos = self.get_pos()
+        self._timestamp = ticks_ms()
+        self.cb = move_callback
+
+        # callback function on events.
+        self.on_move_do = lambda state: self.cb(state)
+
+    def get_state(self):
+        return {
+            'direction': self.is_inverted and -self._direction or self._direction,
+            'position': self.is_inverted and -self._pos or self._pos,
+        }
```

Comment:
In my opinion the return type should be a proper class `PotentiometerState` and not a dictionary.
The encoder module also uses a dict, but I think a class is more pythonic.